<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>418</width>
    <height>562</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Motor Calibrate</string>
  </property>
  <widget class="QWidget" name="formLayoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>418</width>
     <height>561</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="11" column="0">
     <widget class="QListWidget" name="list_commanded"/>
    </item>
    <item row="7" column="1" colspan="2">
     <widget class="QListWidget" name="list_estimated">
      <property name="statusTip">
       <string>Parameters for these drivers will be optimized</string>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::DragDrop</enum>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::MoveAction</enum>
      </property>
     </widget>
    </item>
    <item row="9" column="1" colspan="2">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Sensed Paths</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="13" column="0" colspan="3">
     <widget class="Line" name="line_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="6" column="1" colspan="2">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Estimated Drivers</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QListWidget" name="list_drivers">
      <property name="statusTip">
       <string>Parameters for these drivers will not be optimized</string>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::DragDrop</enum>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::MoveAction</enum>
      </property>
     </widget>
    </item>
    <item row="3" column="1" colspan="2">
     <widget class="QListWidget" name="list_fixed">
      <property name="statusTip">
       <string>Links that are fixed in space</string>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::DragDrop</enum>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::MoveAction</enum>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_7">
      <property name="text">
       <string>Unestimated Drivers</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Fixed Links</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QToolButton" name="toolButton">
      <property name="text">
       <string>Load</string>
      </property>
     </widget>
    </item>
    <item row="14" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Velocity Error Weight</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="Robot">
      <property name="text">
       <string>Robot</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Commanded Paths</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Free Links</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="lbl_robot">
      <property name="text">
       <string>Select Robot</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="17" column="0" colspan="3">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Calibrate</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_2">
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="11" column="1" colspan="2">
     <widget class="QListWidget" name="list_sensed"/>
    </item>
    <item row="4" column="0" colspan="3">
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="3">
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QListWidget" name="list_free">
      <property name="statusTip">
       <string>Drag items to Fixed Links lists to freeze them in space</string>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::DragDrop</enum>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::MoveAction</enum>
      </property>
     </widget>
    </item>
    <item row="8" column="0" colspan="3">
     <widget class="Line" name="line_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="15" column="0">
     <widget class="QLabel" name="label_9">
      <property name="text">
       <string>Timestep</string>
      </property>
     </widget>
    </item>
    <item row="14" column="1" colspan="2">
     <widget class="QLineEdit" name="line_error">
      <property name="text">
       <string>0.0005</string>
      </property>
     </widget>
    </item>
    <item row="16" column="0">
     <widget class="QLabel" name="label_8">
      <property name="text">
       <string>Iterations</string>
      </property>
     </widget>
    </item>
    <item row="16" column="1" colspan="2">
     <widget class="QSpinBox" name="spn_numiters">
      <property name="maximum">
       <number>99999</number>
      </property>
      <property name="value">
       <number>1000</number>
      </property>
     </widget>
    </item>
    <item row="15" column="1" colspan="2">
     <widget class="QLineEdit" name="line_dt">
      <property name="text">
       <string>0.0001</string>
      </property>
     </widget>
    </item>
    <item row="12" column="0" colspan="3">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_4">
        <property name="text">
         <string>Add</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_3">
        <property name="text">
         <string>Delete</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>LoadRobot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>AddPaths()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>75</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>DeletePaths()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>360</x>
     <y>355</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>DoCalibrate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>54</x>
     <y>540</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>276</x>
     <y>537</y>
    </hint>
    <hint type="destinationlabel">
     <x>329</x>
     <y>545</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>list_commanded</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>SetPathIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>373</y>
    </hint>
    <hint type="destinationlabel">
     <x>-4</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>list_sensed</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>SetPathIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>487</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>LoadRobot()</slot>
  <slot>AddPaths()</slot>
  <slot>DeletePaths()</slot>
  <slot>DoCalibrate()</slot>
  <slot>SetPathIndex(int)</slot>
 </slots>
</ui>
